Application/Published/Upgrade-AdminApp/Deploy-EdFiOdsAdminApp.ps1

```diff
@@ -0,0 +1,220 @@
+{
+	$TemplateFileDirectory = $PSScriptRoot;
+}
+
+$CacheTimeOut = "10"
+
+$AdminAppTemplateFile = "$TemplateFileDirectory\OdsAdminApp.Upgrade.json"
+$AdminAppTemplateParametersFile = "$TemplateFileDirectory\OdsAdminApp.parameters.Upgrade.json"
+
+function Get-SqlServerInfo()
+{	
+	$hostName = Read-Host -Prompt "SQL Server Hostname (ex: sql.mydomain.com,1433)"
+	$hostName = $hostName.Replace(':', ',')
+	$adminCredentialsPrompt = "Please enter a username and password for your SQL Server.  These credentials will be used to create new database users for your Ed-Fi ODS installation."
+
+	$sqlServer = @{
+		HostName = $hostName
+		AdminCredentials = (Get-CredentialFromConsole $adminCredentialsPrompt)	
+	}	
+	return $sqlServer
+}
+
+function Deploy-AdminApp($odsDeployInfo)
+{
+	$resourceGroupName = $odsDeployInfo.ResourceGroupName
+	$appInsightsLocation = $odsDeployInfo.AppInsightsLocation	
+	$sqlServer = $odsDeployInfo.SqlServerInfo
+	$productionApiUrl = $odsDeployInfo.ProductionApiUrl	
+
+	if(-not $EncryptionKey)
+	{
+		$aes = [System.Security.Cryptography.Aes]::Create()
+		$aes.KeySize = 256
+		$aes.GenerateKey()
+		$EncryptionKey = [System.Convert]::ToBase64String($aes.Key)
+	}
+	
+	$deployParameters = New-Object -TypeName Hashtable
+	$deployParameters.Add("version", $AdminAppVersion)
+	$deployParameters.Add("edition", $Edition)
+	$deployParameters.Add("appInsightsLocation", $appInsightsLocation)
+	$deployParameters.Add("odsInstanceName", $resourceGroupName)
+	$deployParameters.Add("sqlServerAdminLogin", $sqlServer.AdminCredentials.UserName)
+	$deployParameters.Add("sqlServerAdminPassword", $sqlServer.AdminCredentials.Password)
+	$deployParameters.Add("productionApiUrl", $productionApiUrl)	
+	$deployParameters.Add("metadataCacheTimeOut", $CacheTimeOut)
+	$deployParameters.Add("adminAppNameToDeploy", $AdminAppName)
+	$deployParameters.Add("encryptionKey", $EncryptionKey)
+	
+	$templateFile = $AdminAppTemplateFile
+	$templateParametersFile = $AdminAppTemplateParametersFile
+
+	Try
+	{
+		Write-Host "Deploying ODS Admin App"		
+		$deploymentResult = New-AzureRmResourceGroupDeployment -DeploymentDebugLogLevel All -Name ((Get-ChildItem $templateFile).BaseName + '-' + ((Get-Date).ToUniversalTime()).ToString('MMdd-HHmm')) `
+									   -ResourceGroupName $resourceGroupName `
+									   -TemplateFile $templateFile `
+									   -TemplateParameterFile $templateParametersFile `
+									   @deployParameters `
+									   -Force -Verbose -ErrorAction Stop
+	}
+	Catch
+	{
+		Write-Error $_.Exception.Message		
+	}
+
+	$adminAppUrl = $deploymentResult.Outputs.adminAppUrl.Value
+	Write-Success "Ed-Fi ODS Admin App accessible at $adminAppUrl"
+	
+	return $deploymentResult
+}
+
+function IsDBInstalled([string]$Server, [string]$database)
+{	 
+	$t=Invoke-Sqlcmd -ServerInstance $Server -Username  $user -Password  $pwd -Database "master" -Query "select 1 from sys.databases where name='$database'" -OutputSqlErrors $true 
```

Comment:
This is a very long line, what do you think about refactoring to an argument array?